test/session.cpp

```diff
@@ -225,3 +225,87 @@ TEST_CASE_FIXTURE(RunningSessionTest, "Full Session Life-Cycle")
     check(initial_epoch);
   }
 }
+
+TEST_CASE("Session with X509 Credential")
```

Comment:
It seems like it would be good to test with an RSA public key in the leaf.  Would it be straightforward to generate that test data?

Reply:
hpke/test/certificate.cpp has a test with rsa all through..

Reply:
That just tests *parsing*.  The bug being fixed here is that the public key in the certificate couldn't be used to verify anything.  So that's what you need to test.  You're probably correct that it should be in `lib/hpke/test/certificate.cpp`.  If you don't want to mess with importing private keys, you could just have your test generation script produce an example (message, signature) under the certified public key, and verify that the parsed public key can be used to verify the signature.